docs/cryptography_overview.md

```diff
@@ -0,0 +1,61 @@
+
+### Client being protected against a rogue Worker.
+
+A client wants to be sure that the worker that ends up doing the job the client asked for, is actually a worker that it trusts. Since
+routing is done with a public key, a rogue worker could connect to the scheduler saying "hey, i'm a worker with this key too, send jobs my way!".
+An adversarial worker like that would not be able to decrypt any of the client's request, nor would it be able to send back a result message. The worker doesn't have the private key required to generate the shared secret required to decrypt the clients request. This secret is also necissery to send a response back to the client that the client will be able to decrypt. There might still be a DDOS attack vector here, but no data leak one.
+
+
+### Configuration Example
+
+A program like bonanza_builder is both a worker (you can ask it to do a build for you), but also a client (it can ask a bonanza_worker to execute a job). It uses different keys for these roles. Because it is a worker it's configured with a:
+
+- **platformPrivateKey**: used for routing. (The name platform feels very weird actually, maybe we should rename it to WorkerKindPrivateKey?) You don't have to provide the public key, because it can be generated from the private one. The public one that's used for routing, but you configure it with the private key, because that's used in the communication as described in [Client-Worker Communication](#client-worker-communication-not-trusting-the-scheduler))
+- **clientCertificateAuthority**: the public-key-inside-certificate that has signed all trustworthy clients' public key certificates.
+
+Because bonanza_builder also is a client, it is additionally configured with:
+- **executionClientCertificateChain**: the public-key-certificate that was signed by the certificate authority that configured the bonanza_worker that bonanza_builder wants to ask to execute jobs.
+- **executionClientPrivateKey**: The corresponding private key. Once the public key is accepted, the private key is used to encrypt the client-worker communication as described in [Client-Worker Communication](#client-worker-communication-not-trusting-the-scheduler).
+
+## Reference Notes
+
+### Cryptography Notes for Non-Experts
+
+- All keys you have to provide in Bonanza configuration files are allowed to be X25519 keys or Ed25519 keys.
+- There's a lot more tooling/support for embedding Ed25519 keys inside certificates. However, the trick to make a shared secret described above does not work with Ed25519 keys. Fortunately you can generate a X25519 key from an Ed25519 key, and that is what Bonanza does when it was provided an Ed25519 key, and wants to generate a shared secret. 
```

Comment:
That first sentence, I would consider phrasing it the other way around. Instead of explaining why Ed25519 works, I would start off with going in a bit more detail why X25519 does not work. Maybe something like this?

> There are various practical limitations to embedding X25519 public keys in X.509 certificates. For example, it's not possible to create Certificate Signing Requests for them. OpenSSL is also unable to generate such certificates. This is why we also support Ed25519. [...]